### PR TITLE
fix: correct curvature signal on RPP

### PR DESCRIPTION
**Pull request by afonsofonbraga**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | |
| Primary OS tested on |Ubuntu |
| Robotic platform tested on | Ackermann robot model|

---

## Description of contribution in a few bullet points
* Testing the RPP with an Ackermann robot moving backwards the robot was moving directions randomly not following the global path: 
error
Taking a closer look into the RPP at this line https://github.com/ros-planning/navigation2/blob/80d7d0a080d7401607fd85216f9c450518e7ed42/nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp#L628 the linear_vel is being multiplied by the signal, changing the direction of the curvature at line 
https://github.com/ros-planning/navigation2/blob/80d7d0a080d7401607fd85216f9c450518e7ed42/nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp#L333
My solution was to also multiply at line 333 by the sign to revert the multiplication and not messing with the applyConstraints function. The result is:
correct

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp [hunk 1]
```diff
@@ -330,7 +330,7 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
       costAtPose(pose.pose.position.x, pose.pose.position.y), linear_vel, sign);
 
     // Apply curvature to angular velocity after constraining linear velocity
-    angular_vel = linear_vel * curvature;
+    angular_vel = linear_vel * sign *curvature;
```
Comment (SteveMacenski) [inline nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp:333]:
```suggestion
    angular_vel = sign * linear_vel * curvature;
```

Fixes a linting problem and also a little more semantically clear if the `sign` is in front

**Comment by SteveMacenski** [comment]
Toggling to trigger CI, it didn't for some reason :shrug:

**Comment by SteveMacenski** [comment]
LGTM other than what's mentioned above + CI run

**Comment by afonsofonbraga** [comment]
Sorry by the force on the push command, the commit wasn't signed off. All fixed now.

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Thanks for the patch!

**Pull request closed without merging**